### PR TITLE
Fix all deprecation warnings

### PR DESCRIPTION
**Pull request by JelteF**
This fixes all deprecation warnings that test output was showing. This is in preparation of Django 4.0 support.

### CODE REVIEW

File: django_multitenant/tests/test_models.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def test_filter_select_related(self):
             task = Task.objects.filter(pk=task_id).select_related("project").first()
 
             self.assertEqual(task.account_id, task.project.account_id)
-            pattern = 'INNER JOIN "tests_project" ON \("tests_task"."project_id" = "tests_project"."id" AND \("tests_task"."account_id" = .?"tests_project"."account_id"\)\)'
+            pattern = 'INNER JOIN "tests_project" ON \\("tests_task"."project_id" = "tests_project"."id" AND \\("tests_task"."account_id" = .?"tests_project"."account_id"\\)\\)'
```
Comment (thanodnl) [inline django_multitenant/tests/test_models.py:118]: wait, what? how did this pass before?

File: django_multitenant/tests/urls.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # -*- coding: utf-8 -*-
-from django.conf.urls import include, url
+import django
+from django.conf.urls import include
+
+if django.VERSION >= (3, 1):
```
Comment (thanodnl) [inline django_multitenant/tests/urls.py:5]: above you check `django.VERSION > (3,2)` which seems semantically the same.
Could we standardize on 1 variant of testing this?

I think I like the variant used at this instance better than above, as it tells me slightly more precise what is included.

How does this work with patch versions in `3.1`? (assuming semver)

File: django_multitenant/__init__.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # -*- coding: utf-8 -*-
+import django
+
 version = (2, 3, 2)
 
 __version__ = ".".join(map(str, version))
 
-default_app_config = "django_multitenant.apps.MultitenantConfig"
+if django.VERSION < (3, 2):
+    default_app_config = "django_multitenant.apps.MultitenantConfig"
```
Comment (thanodnl) [inline django_multitenant/__init__.py:9]: assuming this is correct, a comment might be called for. Why don't we have to set `default_app_config` in later versions?

As per earlier comment: lets standardize on a way we version check the difference between 3.1 and 3.2, if possible.

**Review by thanodnl — DISMISSED** [review]

File: django_multitenant/tests/test_models.py (hunk 1 above)
Comment (JelteF) [reply django_multitenant/tests/test_models.py:118]: apparently at the moment escaping a character that isn't recognized as an escape sequence results in a literal `\` in front of it. This was marked as deprecated by some semi-recent python versions. Simple test in python REPL:
```python
>>> "\("
'\\('
```

File: django_multitenant/tests/urls.py (hunk 2 above)
Comment (JelteF) [reply django_multitenant/tests/urls.py:5]: > How does this work with patch versions in 3.1? (assuming semver)

They are tuple compares, which work like this:
```
>>> (3, 1, 0) >= (3, 1)
True
>>> (3, 1, 1) >= (3, 1)
True
>>> (3, 0, 5) >= (3, 1)
False
```

File: django_multitenant/__init__.py (hunk 3 above)
Comment (JelteF) [reply django_multitenant/__init__.py:11]: I'm not sure standardizing will make this more readable. To have the same behavior you would need to prefix the condition with `not` (or have an empty if block and do this in the `else` part):

```python
if not django.VERSION >= (3, 2):
    ....
```

Comment (JelteF) [reply django_multitenant/__init__.py:11]: After looking through the code I do realize that we are inconsistent in more ways in comparing to `django.VERSION`. I'll make a follow up PR to standardize a bit better.

Comment (thanodnl) [reply django_multitenant/__init__.py:11]: > to have the same behavior you would need to prefix the condition with not (or have an empty if block and do this in the else part):

If that is required lets not change it. I guess it was just me assuming they test the same thing, which might be completely a misunderstanding from my side. Am not super familiar with python and the tuple comparisons as you might have concluded from the earlier comment.

**Review by thanodnl — APPROVED** [review]

**Pull request merged**